inc/vcf/normalizer.hpp

```diff
@@ -32,12 +32,22 @@ namespace ebi
         size_t position;
         std::string reference_allele;
         std::string alternate_allele;
+        bool is_symbolic_allele;
         
         RecordCore(size_t line, const std::string &chromosome, size_t position, 
                    const std::string &reference_allele, const std::string &alternate_alleles) 
             : line(line), chromosome(chromosome), position(position), 
               reference_allele(reference_allele), alternate_allele(alternate_alleles)
-        { }
+        {
+            // checks the alternate allele to see if it is a symbolic allele.
+            // For this, enclosure by angular brackets is the necessary and sufficient condition
+            if ( (alternate_alleles.find("<") == 0) && (alternate_alleles.find(">") == alternate_alleles.length() - 1) ) {
```

Comment:
what we meant is using the property from Record as another parameter to this constructor

Reply:
Hmm yes, it's already implemented in the Record.

And also if the initial idea (no longer required) was to see if the string `alternate_allele` is enclosed by `<>`, this line wouldn't be ideal. Because the find always returns the first occurrence of a character, and in this case you might get `>` before itself (not sure though, maybe the validator flags it wrong before reaching this block of code?) and not at the last index.

Moreover, you can simply check it by seeing if the character at the first index is `<` and the one at the last one is `>`, I think😅. Why go for O(n) when you can do it in O(1)?

Reply:
Since the tags for all possible structural variants were enclosed within <> and were devoid of angle brackets inside the tag itself, I figured if > is present inside the tag that should be acknowledged as an error.
But I noticed that the Record class uses only the starting angle bracket to identify it as a structural variant, so I guess such errors won't exist at this block of code. And in such a situation, your point about the time complexity is correct.